README.md

```diff
@@ -12,17 +12,70 @@ There are **Primitive Rules** and **Composed Rules** that are created by combini
 
 **Composed Rules** return a true or false based on whether any or all of the primitive rules return true.  Composed rules can then be combined into higher-level composed rules that implement more complex boolean logic.  Because of the recursive definition of the `Rule` enum, calling `validate()` on a top-level composed rule will start at the top and validate at every level, down to the component primitive rules.
 
-# Examples
+## Environment Setup
+1. Install Rust from https://rustup.rs/
+2. Install Solana from https://docs.solana.com/cli/install-solana-cli-tools#use-solanas-install-tool
+3. Run `yarn install` to install dependencies
+
+---
+
+### Build and test the Rust program
+```
+$ cd program/
+$ cargo build-bpf
```

Comment:
FWIW, you don't need to build it if you're just testing. `cargo test-bpf` will build everything it needs to run the tests.

Reply:
yes I know this I just thought it was more beginner friendly to list them as separate steps.  But maybe its redundant.